hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/FreezeUnfreezeTokenPrecompileSuite.java

```diff
@@ -401,4 +427,296 @@ public HapiSpec createNFTTokenFreezeKeyFromHollowAccountAlias() {
+                                NO_KEY_FREEZE_TXN,
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith()
+                                        .status(TOKEN_HAS_NO_FREEZE_KEY)
+                                        .contractCallResult(resultWith()
+                                                .contractCallResult(
+                                                        htsPrecompileResult().withStatus(TOKEN_HAS_NO_FREEZE_KEY)))),
+                        childRecordsCheck(
+                                NO_KEY_UNFREEZE_TXN,
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith()
+                                        .status(TOKEN_HAS_NO_FREEZE_KEY)
+                                        .contractCallResult(resultWith()
+                                                .contractCallResult(
+                                                        htsPrecompileResult().withStatus(TOKEN_HAS_NO_FREEZE_KEY)))));
+    }
+
+    @HapiTest
+    final HapiSpec freezeUnfreezeNFTsWithNegativeCases() {
+        final var IS_FROZEN_TXN = "IS_FROZEN_TXN";
+        final AtomicReference<TokenID> vanillaTokenID = new AtomicReference<>();
+        final AtomicReference<AccountID> accountID = new AtomicReference<>();
+        return defaultHapiSpec("freezeUnfreezeNftsWithNegativeCases")
+                .given(
+                        newKeyNamed(FREEZE_KEY),
+                        newKeyNamed(ADMIN_KEY),
+                        newKeyNamed(MULTI_KEY),
+                        cryptoCreate(ACCOUNT).balance(100 * ONE_HBAR).exposingCreatedIdTo(accountID::set),
+                        cryptoCreate(TOKEN_TREASURY),
+                        tokenCreate(KNOWABLE_TOKEN)
+                                .tokenType(NON_FUNGIBLE_UNIQUE)
+                                .treasury(TOKEN_TREASURY)
+                                .adminKey(ADMIN_KEY)
+                                .freezeKey(FREEZE_KEY)
+                                .supplyKey(MULTI_KEY)
+                                .initialSupply(0)
+                                .exposingCreatedIdTo(id -> vanillaTokenID.set(asToken(id))),
+                        mintToken(KNOWABLE_TOKEN, List.of(copyFromUtf8("First!"))),
+                        uploadInitCode(FREEZE_CONTRACT),
+                        contractCreate(FREEZE_CONTRACT),
+                        tokenAssociate(ACCOUNT, KNOWABLE_TOKEN),
+                        cryptoTransfer(movingUnique(KNOWABLE_TOKEN, 1L).between(TOKEN_TREASURY, ACCOUNT)),
+                        newKeyNamed(THRESHOLD_KEY).shape(THRESHOLD_KEY_SHAPE.signedWith(sigs(ON, FREEZE_CONTRACT))),
+                        tokenUpdate(KNOWABLE_TOKEN).freezeKey(THRESHOLD_KEY).signedByPayerAnd(ADMIN_KEY))
+                .when(withOpContext((spec, opLog) -> allRunFor(
+                        spec,
+                        //                        contractCall(
```

Comment:
did you mean to comment out this test?  If so perhaps we can remove this and the corresponding `then` section below?

Reply:
It is commented out because of a bug that I have found. 
I have described it in `services-modularization` chanel on Friday. You can have a peek there.

Reply:
@lukelee-sl  the test is enabled.